### PR TITLE
Merge Sufia configuration with the CurationConcerns

### LINKED ISSUE #2139: Generator should remove config/initializer/curation_concerns.rb
But we must call
```ruby
CurationConcerns.configure.register_curation_concern :generic_work
```
in some way.  Perhaps Sufia config can delegate that method to CurationConcerns?

### PR DESCRIPTION
**Pull request by jcoyne**
@projecthydra/sufia-code-reviewers

This reduces confusion over which configuration someone should be using.
Fixes #2139

### CODE REVIEW

File: lib/generators/sufia/templates/config/sufia.rb [hunk 1]
```diff
@@ -1,16 +1,11 @@
 Sufia.config do |config|
-  config.max_days_between_audits = 7
-  config.max_notifications_for_dashboard = 5
+  config.register_curation_concern :generic_work
 
-  config.permission_levels = {
-    "Choose Access" => "none",
-    "View/Download" => "read",
-    "Edit" => "edit"
-  }
```
Comment (awead) [inline lib/generators/sufia/templates/config/sufia.rb]: do these just not do anything anymore?

**Comment by awead** [comment]
@jcoyne looks good. Do you want me to merge https://github.com/projecthydra/curation_concerns/pull/848 first so you can go ahead and remove that other stuff in this PR?

File: lib/generators/sufia/templates/config/sufia.rb (hunk 1 above)
Comment (jcoyne) [inline lib/generators/sufia/templates/config/sufia.rb]: I just moved them into Configuration like all the other defaults.

**Comment by jcoyne** [comment]
Naw, we can merge this now and defer removing those methods.

**Comment by awead** [comment]
Works for me. Travis is complaining, but that error appears to be intermittent. I wasn't able to reproduce it locally.

**Pull request merged**